spark-indexer/src/main/java/org/uniprot/store/spark/indexer/proteome/reader/ProteomeStatisticsReader.java

```diff
@@ -0,0 +1,111 @@
+package org.uniprot.store.spark.indexer.proteome.reader;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.api.java.function.Function2;
+import org.uniprot.core.flatfile.parser.impl.cc.CcLineTransformer;
+import org.uniprot.core.proteome.ProteomeStatistics;
+import org.uniprot.core.proteome.impl.ProteomeStatisticsBuilder;
+import org.uniprot.core.uniprotkb.UniProtKBEntry;
+import org.uniprot.core.uniprotkb.UniProtKBEntryType;
+import org.uniprot.core.uniprotkb.comment.Comment;
+import org.uniprot.core.uniprotkb.impl.UniProtKBEntryBuilder;
+import org.uniprot.store.spark.indexer.common.JobParameter;
+import org.uniprot.store.spark.indexer.uniprot.UniProtKBRDDTupleReader;
+import org.uniprot.store.spark.indexer.uniprot.converter.UniProtEntryConverterUtil;
+
+import scala.Tuple2;
+
+public class ProteomeStatisticsReader {
+    public static final String PROTEOMES = "Proteomes";
+    private final UniProtKBRDDTupleReader uniProtKBReader;
+
+    public ProteomeStatisticsReader(JobParameter parameter) {
+        this.uniProtKBReader = new UniProtKBRDDTupleReader(parameter, false);
+    }
+
+    public JavaPairRDD<String, ProteomeStatistics> getProteomeStatisticsRDD() {
+        return JavaPairRDD.fromJavaRDD(
+                        getProteinInfo().map(getProteomeStatistics()).flatMap(List::iterator))
```

Comment:
I don't think lambda expression or function is allowed mapping in spark distributed. Based on my understanding, the mapper should be a serialisable class so that spark can run on dsitributed system.
Also, i don't think .map(method) will work. It should be a class. e.g. .map(new Mapper())

Reply:
we are going to try this on spark, assuming this should be ok